packages/compass-sidebar/src/components/sidebar-title.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { useCallback, useMemo } from 'react';
+import { connect } from 'react-redux';
+import type { ConnectedProps } from 'react-redux';
+import { getConnectionTitle } from 'mongodb-data-service';
+import type { ConnectionInfo } from 'mongodb-data-service';
+import { globalAppRegistryEmit } from '@mongodb-js/mongodb-redux-common/app-registry';
+
+import {
+  MongoDBLogoMark,
+  css,
+  cx,
+  uiColors,
+  spacing,
+  ItemActionControls,
+} from '@mongodb-js/compass-components';
+
+import type { ItemAction } from '@mongodb-js/compass-components';
+
+type Actions = 'copy-connection-string' | 'edit-favorite' | 'cluster-info';
+
+const titleContainer = css({
+  // TODO: move, also we need dark colors too
+  '--title-color': 'white',
+  '--title-bg-color': uiColors.green.dark2,
+
+  '--icon-color': 'white',
+  // TODO: ok so using the hover version makes no sense and we do have to actually specify "normal" tpo ItemActionControls
+  '--hover-icon-color': 'white',
+  // TODO: we also need a -hover foreground colour in this case
+
+  display: 'flex',
+  alignItems: 'center',
+  cursor: 'pointer',
+  color: 'var(--title-color)',
+  backgroundColor: 'var(--title-bg-color)',
+
+  height: spacing[6] + spacing[1],
+  padding: spacing[3],
+});
+
+const titleLabel = css({
+  overflow: 'hidden',
+  whiteSpace: 'nowrap',
+  textOverflow: 'ellipsis',
+  fontSize: '18px', // TODO: what's the right way to do this?
+  fontWeight: 600, // TODO: 500 once we have the new font
+  marginLeft: '2px',
+  paddingRight: spacing[2],
+});
+
+const TitleLabel: React.FunctionComponent<React.HTMLProps<HTMLSpanElement>> = ({
+  children,
+  className,
+  ...props
+}) => {
+  return (
+    <span className={cx(titleLabel, className)} {...props}>
+      {children}
+    </span>
+  );
+};
+
+const logoContainer = css({
+  width: spacing[5],
+  paddingLeft: spacing[1],
+  marginTop: '6px',
+  flexShrink: 0,
+});
+
+function TitleLogo() {
+  return (
+    <div className={logoContainer}>
+      <MongoDBLogoMark color="green-base" height={32} />
+    </div>
+  );
+}
+
+const mapStateToProps = (
+  state: any
+): {
+  activeNamespace: string;
+  connectionInfo: ConnectionInfo;
+} => ({
+  activeNamespace: state.databases.activeNamespace,
+  connectionInfo: state.connectionInfo.connectionInfo,
+});
+
+const connector = connect(mapStateToProps, {
+  globalAppRegistryEmit,
+});
+
+type PropsFromRedux = ConnectedProps<typeof connector>;
```

Comment:
I know that it's a draft and you'd probably iterate more on this I would still want to highlight that I would recommend not defining component props based on connect mappers: when you do it that way you unintentionally create a very strong coupling between state and component. I would recommend going from the component implementation and looking for the minimal set of props required and then implementing connect to match it. In this case for example it seems like the only thing you want from connectionInfo is title, and so the component interface can be simplified and the `getConnectionTitle` part can be moved to the state mapper

Reply:
Thinking about this more: Should I connect at all? might as well just pass in title, favorite and onAction from the sidebar?

Reply:
Generally speaking redux docs [strongly suggest](https://redux.js.org/style-guide/#connect-more-components-to-read-data-from-the-store) to connect as much components as possible and provide the data as granular as possible. I think this advice works great in most cases, even when it doesn't help much with rendering performance, but absolutely best when your whole app is following this. I'd say in this case it depends whether or not you are already using connection info and other connected props / methods somewhere a few levels higher in the react tree. If you do, then yeah, I think that would be fine to not connect